### PR TITLE
Implement minimal Visual Basic Application Framework WinForms app.

### PR DESCRIPTION
**Pull request by KlausLoeffelmann**
This PR adds a minimal Visual Basic WinForms app project, which uses the Visual Basic Application Framework, to the test folder.

**Note:** This app is to have a central lookup spot how to activate the application framework for Visual Basic WinForms net 5.0 apps as long as there isn't a UI to setup the Application Framework parameters.

**Issues:**
* HighDpiMode is setup in the StartUp Application Event.
* SingleInstance cannot be set, since this is currently not supported.
* Changes of the Application Framework Settings can only be made directly by patching the _My project\Application.Designer.vb_ file. Changes to _Application.myapp_ don't have any effect.

[UPDATE:] @merriemcgaw, @OliaG, @RussKie , @paul1956, @davkean
* Addressed most of the review suggestions.
* SingleInstance is working now for .NET 5.0 Apps.
* Changes in Application.myapp trigger code generation on Application.Designer.Vb on save.

###### Microsoft Reviewers: [Open in CodeFlow](http://wpcp.azurewebsites.net/CodeFlowProtocolProxy2.php?pullrequest=https://github.com/dotnet/winforms/pull/3352)

### CODE REVIEW

File: src/System.Windows.Forms/tests/IntegrationTests/VisualBasic AppFramework Net5/My Project/AssemblyInfo.vb [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+<Assembly: AssemblyDescription("")>
+<Assembly: AssemblyCompany("")>
+<Assembly: AssemblyProduct("VbWinForms")>
+<Assembly: AssemblyCopyright("Copyright ©  2020")>
+<Assembly: AssemblyTrademark("")>
+
+<Assembly: ComVisible(False)>
+
+'The following GUID is for the ID of the typelib if this project is exposed to COM
+<Assembly: Guid("6a6ec69e-e4b7-4dcb-8a58-3e82006838ce")>
```
Comment (RussKie) [inline src/System.Windows.Forms/tests/IntegrationTests/VisualBasic AppFramework Net5/My Project/AssemblyInfo.vb:21]: Are these needed?

File: src/System.Windows.Forms/tests/IntegrationTests/VisualBasic AppFramework Net5/VbWinForms.vbproj [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+    <ItemGroup>
+        <Import Include="System.Data" />
+        <Import Include="System.Drawing" />
+        <Import Include="System.Windows.Forms" />
+    </ItemGroup>
+
+    <ItemGroup>
+	    <None Include="My Project\Application.Designer.vb">
+		    <DependentUpon>Application.myapp</DependentUpon>
+	    </None>
```
Comment (RussKie) [inline src/System.Windows.Forms/tests/IntegrationTests/VisualBasic AppFramework Net5/VbWinForms.vbproj:32]: Please convert all tabs to spaces

File: src/System.Windows.Forms/tests/IntegrationTests/VisualBasic AppFramework Net5/My Project/AssemblyInfo.vb (hunk 1 above)
Comment (paul1956) [reply src/System.Windows.Forms/tests/IntegrationTests/VisualBasic AppFramework Net5/My Project/AssemblyInfo.vb:21]: This GUID is the one in Framework that is used as **part** of the UniqueID for Single Instance. It comes from a field in the Visual Studio UI that is no longer there. Currently I have not found any replacement if you want the same behavior as Framework.

Comment (paul1956) [reply src/System.Windows.Forms/tests/IntegrationTests/VisualBasic AppFramework Net5/My Project/AssemblyInfo.vb:21]: If this GUID is a constant I can imagine one test/prototype app killing another because someone forgets to change from constant value shown. If it is dynamic then we have the same issue we have with the prototype Single Instance where it changes on every build and is different between Debug and Release.

File: src/System.Windows.Forms/tests/IntegrationTests/VisualBasic AppFramework Net5/My Project/AssemblyInfo.vb [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+Imports System.Reflection
+Imports System.Runtime.InteropServices
+
+' General Information about an assembly is controlled through the following
+' set of attributes. Change these attribute values to modify the information
+' associated with an assembly.
+
+' Review the values of the assembly attributes
+
+<Assembly: AssemblyTitle("VbWinForms")>
```
Comment (paul1956) [inline src/System.Windows.Forms/tests/IntegrationTests/VisualBasic AppFramework Net5/My Project/AssemblyInfo.vb:11]: I don't believe AssemblyTitle has a Visual Studio UI under core either.

**Comment by paul1956** [comment]
I think there is also a need to set NeutralResourcesLanguage otherwise you get CA1824. for example
```
<Assembly: NeutralResourcesLanguage("en-US")>
```

File: src/System.Windows.Forms/tests/IntegrationTests/VisualBasic AppFramework Net5/My Project/AssemblyInfo.vb (hunk 3 above)
Comment (paul1956) [inline src/System.Windows.Forms/tests/IntegrationTests/VisualBasic AppFramework Net5/My Project/AssemblyInfo.vb:11]: Using your example above the Title should be "VB WinForms NET5.0"

File: src/System.Windows.Forms/tests/IntegrationTests/VisualBasic AppFramework Net5/ApplicationEvents.vb [hunk 4]
```diff
@@ -0,0 +1,16 @@
...
+
+Namespace My
+    ' The following events are available for MyApplication:
+    ' Startup: Raised when the application starts, before the startup form is created.
+    ' Shutdown: Raised after all application forms are closed.  This event is not raised if the application terminates abnormally.
+    ' UnhandledException: Raised if the application encounters an unhandled exception.
+    ' StartupNextInstance (!!Not yet implemented!!): Raised when launching a single-instance application and the application is already active. 
+    ' NetworkAvailabilityChanged: Raised when the network connection is connected or disconnected.
+    Partial Friend Class MyApplication
+        Private Sub MyApplication_Startup(sender As Object, e As StartupEventArgs) Handles Me.Startup
```
Comment (paul1956) [inline src/System.Windows.Forms/tests/IntegrationTests/VisualBasic AppFramework Net5/ApplicationEvents.vb:12]: All 5 events should be included since VS will not offer to create the file if it already exists.
```VB
' Licensed to the .NET Foundation under one or more agreements.
' The .NET Foundation licenses this file to you under the MIT license.
' See the LICENSE file in the project root for more information.

Imports Microsoft.VisualBasic.ApplicationServices
Imports AppFramework = System.Windows.Forms.Application

Namespace My
    ' The following events are available for MyApplication:
    ' Startup: Raised when the application starts, before the startup form is created.
    ' Shutdown: Raised after all application forms are closed.  This event is not raised if the application terminates abnormally.
    ' UnhandledException: Raised if the application encounters an unhandled exception.
    ' StartupNextInstance: Raised when launching a single-instance application and the application is already active. 
    ' NetworkAvailabilityChanged: Raised when the network connection is connected or disconnected.
    Partial Friend Class MyApplication
        Private Sub MyApplication_Startup(sender As Object, e As StartupEventArgs) Handles Me.Startup
            AppFramework.SetHighDpiMode(HighDpiMode.SystemAware)
        End Sub

        'Private Sub MyApplication_Shutdown(sender As Object, e As EventArgs) Handles Me.Shutdown
        'End Sub

        'Private Sub MyApplication_UnhandledException(sender As Object, e As UnhandledExceptionEventArgs) Handles Me.UnhandledException
        'End Sub

        'Public Sub MyApplication_StartupNextInstance(sender As Object, e As StartupNextInstanceEventArgs) Handles Me.StartupNextInstance
        'End Sub

        'Private Sub MyApplication_NetworkAvailabilityChanged(sender As Object, e As Devices.NetworkAvailableEventArgs) Handles Me.NetworkAvailabilityChanged
        'End Sub
    End Class
End Namespace

```

File: src/System.Windows.Forms/tests/IntegrationTests/VisualBasic AppFramework Net5/VbWinForms.vbproj (hunk 2 above)
Comment (paul1956) [reply src/System.Windows.Forms/tests/IntegrationTests/VisualBasic AppFramework Net5/VbWinForms.vbproj:32]: This may not be the place to report this but where is My.Application? Below may not be complete but is needed for About.vb and SplashScreen.vb to work. 
```
Imports Microsoft.VisualBasic.ApplicationServices
Namespace My
    Partial Public Module Application
        Private s_info As AssemblyInfo
        Public ReadOnly Property Info As AssemblyInfo
            Get
                If s_info Is Nothing Then
                    s_info = New AssemblyInfo(GetType(Form1).Assembly)
                End If
                Return s_info
            End Get
        End Property
    End Module
End Namespace
```

File: src/System.Windows.Forms/tests/IntegrationTests/VisualBasic AppFramework Net5/My Project/AssemblyInfo.vb (hunk 1 above)
Comment (KlausLoeffelmann) [reply src/System.Windows.Forms/tests/IntegrationTests/VisualBasic AppFramework Net5/My Project/AssemblyInfo.vb:21]: We will address this in the template work.
For this scope of this sample, it's not relevant, but!
If folks take this as a template, they should be aware to change this `Guid` for every new project.
Adding a comment in the code.

File: src/System.Windows.Forms/tests/IntegrationTests/VisualBasic AppFramework Net5/VbWinForms.vbproj (hunk 2 above)
Comment (KlausLoeffelmann) [reply src/System.Windows.Forms/tests/IntegrationTests/VisualBasic AppFramework Net5/VbWinForms.vbproj:32]: > Please convert all tabs to spaces
Sure.

Comment (KlausLoeffelmann) [reply src/System.Windows.Forms/tests/IntegrationTests/VisualBasic AppFramework Net5/VbWinForms.vbproj:32]: @paul1956: Thanks for pointing this out! 
It's beyond the scope of this particular sample, but may become relevant for the template work.
A question though: If you create a new WinForms app in NetFx, do you already see this then, or only if you add other templates? (Is it a separate file - other than Application.Designer.vb?)

File: src/System.Windows.Forms/tests/IntegrationTests/VisualBasic AppFramework Net5/My Project/AssemblyInfo.vb (hunk 3 above)
Comment (KlausLoeffelmann) [reply src/System.Windows.Forms/tests/IntegrationTests/VisualBasic AppFramework Net5/My Project/AssemblyInfo.vb:11]: Visual Studio work is still in progress.
So, for this sample, it is not immediate relevant.

**Comment by KlausLoeffelmann** [comment]
> I think there is also a need to set NeutralResourcesLanguage otherwise you get CA1824. for example

@paul1956 : We don't see any warnings. Can you do a repro for this?

**Comment by KlausLoeffelmann** [comment]
@paul1956: Sorry addressing this took a while. 
We also want to explicitly thank you for your support on this!

File: src/System.Windows.Forms/tests/IntegrationTests/VisualBasic AppFramework Net5/ApplicationEvents.vb (hunk 4 above)
Comment (KlausLoeffelmann) [reply src/System.Windows.Forms/tests/IntegrationTests/VisualBasic AppFramework Net5/ApplicationEvents.vb:12]: Done.

File: src/System.Windows.Forms/tests/IntegrationTests/VisualBasic AppFramework Net5/My Project/AssemblyInfo.vb (hunk 3 above)
Comment (KlausLoeffelmann) [reply src/System.Windows.Forms/tests/IntegrationTests/VisualBasic AppFramework Net5/My Project/AssemblyInfo.vb:11]: Changed.

File: src/System.Windows.Forms/tests/IntegrationTests/VisualBasic AppFramework Net5/VbWinForms.vbproj (hunk 2 above)
Comment (paul1956) [reply src/System.Windows.Forms/tests/IntegrationTests/VisualBasic AppFramework Net5/VbWinForms.vbproj:33]: If you add About or Splashscreen, or they come from existing Framework Application they reference "My.Application.Assembly". Application.Designer.vb defines My.MyApplication not My.Application. If you hover over My.Application is says "Readonly Property My.MyProject.Applicaion As My.MyApplication"
Below is typical use
```VB
ApplicationTitle.Text = My.Application.Info.Title
```
If you go to definition of Info you get
```
#Region "Assembly Microsoft.VisualBasic, Version=10.0.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a"
' C:\Program Files (x86)\Reference Assemblies\Microsoft\Framework\.NETFramework\v4.7.2\Microsoft.VisualBasic.dll
#End Region

Imports System.Globalization
Imports Microsoft.VisualBasic.Logging

Namespace Microsoft.VisualBasic.ApplicationServices
    '
    ' Summary:
    '     Provides properties, methods, and events related to the current application.
    Public Class ApplicationBase
        '
        ' Summary:
        '     Initializes a new instance of the Microsoft.VisualBasic.ApplicationServices.ApplicationBase
        '     class.
        Public Sub New()

        '
        ' Summary:
        '     Gets an object that provides properties and methods for writing event and exception
        '     information to the application's log listeners.
        '
        ' Returns:
        '     The Microsoft.VisualBasic.Logging.Log object for the current application.
        Public ReadOnly Property Log As Log
        '
        ' Summary:
        '     Gets an object that provides properties for getting information about the application's
        '     assembly, such as the version number, description, and so on.
        '
        ' Returns:
        '     The Microsoft.VisualBasic.ApplicationServices.AssemblyInfo object for the current
        '     application.
        Public ReadOnly Property Info As AssemblyInfo
        '
        ' Summary:
        '     Gets the culture that the current thread uses for string manipulation and string
        '     formatting.
        '
        ' Returns:
        '     A System.Globalization.CultureInfo object that represents the culture the current
        '     thread uses for string manipulation and string formatting.
        Public ReadOnly Property Culture As CultureInfo
        '
        ' Summary:
        '     Gets the culture that the current thread uses for retrieving culture-specific
        '     resources.
        '
        ' Returns:
        '     A System.Globalization.CultureInfo object that represents the culture that the
        '     current thread uses for retrieving culture-specific resources.
        Public ReadOnly Property UICulture As CultureInfo

        '
        ' Summary:
        '     Changes the culture used by the current thread for string manipulation and for
        '     string formatting.
        '
        ' Parameters:
        '   cultureName:
        '     String. Name of the culture as a string. For a list of possible names, see System.Globalization.CultureInfo.
        '
        ' Exceptions:
        '   T:System.ArgumentNullException:
        '     cultureName is Nothing.
        '
        '   T:System.ArgumentException:
        '     cultureName is not a valid culture name.
        Public Sub ChangeCulture(cultureName As String)
        '
        ' Summary:
        '     Changes the culture that the current thread uses for retrieving culture-specific
        '     resources.
        '
        ' Parameters:
        '   cultureName:
        '     String. Name of the culture as a string. For a list of possible names, see System.Globalization.CultureInfo.
        '
        ' Exceptions:
        '   T:System.ArgumentNullException:
        '     cultureName is Nothing.
        '
        '   T:System.ArgumentException:
        '     cultureName is not a valid culture name.
        Public Sub ChangeUICulture(cultureName As String)

        '
        ' Summary:
        '     Returns the value of the specified environment variable.
        '
        ' Parameters:
        '   name:
        '     A String containing the name of the environment variable.
        '
        ' Returns:
        '     A String containing the value of the environment variable with the name name.
        '
        ' Exceptions:
        '   T:System.ArgumentNullException:
        '     name is Nothing.
        '
        '   T:System.ArgumentException:
        '     The environment variable specified by name does not exist.
        '
        '   T:System.Security.SecurityException:
        '     The calling code does not have System.Security.Permissions.EnvironmentPermission
        '     with Read access.
        Public Function GetEnvironmentVariable(name As String) As String
    End Class
End Namespace
```

**Comment by paul1956** [comment]
@KlausLoeffelmann  CA1824 comes from RoslynAnalyzers, so it may be out of scope or it may be a Visual Studio issue which is where it is set in Framework.

File: src/System.Windows.Forms/tests/IntegrationTests/VisualBasic AppFramework Net5/ApplicationEvents.vb (hunk 4 above)
Comment (paul1956) [reply src/System.Windows.Forms/tests/IntegrationTests/VisualBasic AppFramework Net5/ApplicationEvents.vb:12]: I don't see the menu above in Form1 can you provide some context on how I get to them, these are not Form Events they are connected to My.Application.

Today they are in a different file  "ApplicationEvents.vb" that VS installs when you click on "View Application Events" I was not aware of this feature.
image

File: src/System.Windows.Forms/tests/IntegrationTests/VisualBasic AppFramework Net5/VbWinForms.vbproj [hunk 5]
```diff
@@ -0,0 +1,50 @@
...
+    <PropertyGroup>
+        <OptionExplicit>On</OptionExplicit>
+        <OptionCompare>Binary</OptionCompare>
+        <OptionStrict>On</OptionStrict>
+        <OptionInfer>On</OptionInfer>
+    </PropertyGroup>
+
+    <PropertyGroup>
+        <MyType>WindowsForms</MyType>
+        <!--<ApplicationManifest>My Project\app.manifest</ApplicationManifest>-->
```
Comment (JeremyKuhne) [inline src/System.Windows.Forms/tests/IntegrationTests/VisualBasic AppFramework Net5/VbWinForms.vbproj:21]: Why is this here?

**Comment by paul1956** [comment]
It would be fantastic if for VB nothing was hidden, this was brought up by @KathleenDollard and this seems to be heading that way. By "nothing hidden" I mean the files get generated as part of the project source and not just in files hidden in "bin". For example the command below would update ApplicationInfo.vb instead of generating a new file somewhere and if ApplicationInfo.vb exists get duplicate references.
```
<GenerateAssemblyInfo>true</GenerateAssemblyInfo>
```

File: src/System.Windows.Forms/tests/IntegrationTests/VisualBasic AppFramework Net5/ApplicationEvents.vb (hunk 4 above)
Comment (KlausLoeffelmann) [reply src/System.Windows.Forms/tests/IntegrationTests/VisualBasic AppFramework Net5/ApplicationEvents.vb:12]: They are not in Form1, but in ApplicationEvent.vb.
This file gets inserted in the classic designer with View Application Events, should it not exist yet.
But that is Project System work, and not scope of this PR.

Comment (paul1956) [reply src/System.Windows.Forms/tests/IntegrationTests/VisualBasic AppFramework Net5/ApplicationEvents.vb:12]: @KlausLoeffelmann that is what I thought but your picture shows Form1. If you create the file because you need to set SetHighDpiMode than Visual Studio can add the rest as you show and no extra work is needed.

File: src/System.Windows.Forms/tests/IntegrationTests/VisualBasic AppFramework Net5/VbWinForms.vbproj (hunk 5 above)
Comment (KlausLoeffelmann) [reply src/System.Windows.Forms/tests/IntegrationTests/VisualBasic AppFramework Net5/VbWinForms.vbproj:21]: We can get rid of this.

File: src/System.Windows.Forms/tests/IntegrationTests/VisualBasic AppFramework Net5/ApplicationEvents.vb (hunk 4 above)
Comment (KlausLoeffelmann) [reply src/System.Windows.Forms/tests/IntegrationTests/VisualBasic AppFramework Net5/ApplicationEvents.vb:12]: Well, the screenshot is actually captured unfortunate. ApplicationEvents.vb was the active file at the time.

We need to talk about SetHighDpiMode, anyway. Ideal would be an additional option, so the code for SetHighDpiMode would be created based on a related settings. But also not part of the scope of this.

Comment (paul1956) [reply src/System.Windows.Forms/tests/IntegrationTests/VisualBasic AppFramework Net5/ApplicationEvents.vb:12]: I see an advantage of always creating ApplicationEvents.VB again with the benefits of not hiding stuff. Then VS will help adding the additional Events as your image shows, though I am not sure how many people know it does this, I didn't. I always used the VS button and the cut and paste from Docs.

File: src/System.Windows.Forms/tests/IntegrationTests/VisualBasic AppFramework Net5/My Project/AssemblyInfo.vb [hunk 6]
```diff
@@ -0,0 +1,36 @@
+﻿Imports System
```
Comment (davkean) [inline src/System.Windows.Forms/tests/IntegrationTests/VisualBasic AppFramework Net5/My Project/AssemblyInfo.vb:1]: Why aren't we using auto-gen of AssemblyInfo.cs?

File: src/System.Windows.Forms/tests/IntegrationTests/VisualBasic AppFramework Net5/VbWinForms.vbproj [hunk 7]
```diff
@@ -0,0 +1,49 @@
+﻿<Project Sdk="Microsoft.NET.Sdk.WindowsDesktop">
+    <PropertyGroup>
+        <OutputType>WinExe</OutputType>
+        <TargetFramework>net5.0</TargetFramework>
+        <RootNamespace>VbWinForms</RootNamespace>
+	    <StartupObject>VbWinForms.My.MyApplication</StartupObject>
+	    <GenerateAssemblyInfo>false</GenerateAssemblyInfo>
+        <UseWindowsForms>true</UseWindowsForms>
+        <LangVersion>16</LangVersion>
```
Comment (davkean) [inline src/System.Windows.Forms/tests/IntegrationTests/VisualBasic AppFramework Net5/VbWinForms.vbproj:9]: LangVersion is dictated based on TFM, we shouldn't have this.

File: src/System.Windows.Forms/tests/IntegrationTests/VisualBasic AppFramework Net5/VbWinForms.vbproj [hunk 8]
```diff
@@ -0,0 +1,49 @@
...
+    </PropertyGroup>
+
+    <PropertyGroup>
```
Comment (davkean) [inline src/System.Windows.Forms/tests/IntegrationTests/VisualBasic AppFramework Net5/VbWinForms.vbproj:12]: These properties should be defaulted in the SDK, are they not?

File: src/System.Windows.Forms/tests/IntegrationTests/VisualBasic AppFramework Net5/VbWinForms.vbproj [hunk 9]
```diff
@@ -0,0 +1,49 @@
+﻿<Project Sdk="Microsoft.NET.Sdk.WindowsDesktop">
+    <PropertyGroup>
+        <OutputType>WinExe</OutputType>
+        <TargetFramework>net5.0</TargetFramework>
+        <RootNamespace>VbWinForms</RootNamespace>
```
Comment (davkean) [inline src/System.Windows.Forms/tests/IntegrationTests/VisualBasic AppFramework Net5/VbWinForms.vbproj:5]: RootNamespace should be above `<TargetFramework>` to match the Console.

File: src/System.Windows.Forms/tests/IntegrationTests/VisualBasic AppFramework Net5/VbWinForms.vbproj [hunk 10]
```diff
@@ -0,0 +1,49 @@
...
+	    <StartupObject>VbWinForms.My.MyApplication</StartupObject>
```
Comment (davkean) [inline src/System.Windows.Forms/tests/IntegrationTests/VisualBasic AppFramework Net5/VbWinForms.vbproj:6]: Is this actually needed or do we pick up the default automatically?

File: src/System.Windows.Forms/tests/IntegrationTests/VisualBasic AppFramework Net5/VbWinForms.vbproj [hunk 11]
```diff
@@ -0,0 +1,49 @@
...
+    <ItemGroup>
+      <Compile Update="My Project\Application.Designer.vb">
+        <DesignTime>True</DesignTime>
+        <AutoGen>True</AutoGen>
+        <DependentUpon>Application.myapp</DependentUpon>
+      </Compile>
+    </ItemGroup>
+
+    <ItemGroup>
+      <None Update="My Project\Application.myapp">
```
Comment (davkean) [inline src/System.Windows.Forms/tests/IntegrationTests/VisualBasic AppFramework Net5/VbWinForms.vbproj:44]: This should be defaulted in the SDK, not the template.

File: src/System.Windows.Forms/tests/IntegrationTests/VisualBasic AppFramework Net5/VbWinForms.vbproj [hunk 12]
```diff
@@ -0,0 +1,49 @@
...
+        <OptionCompare>Binary</OptionCompare>
+        <OptionStrict>On</OptionStrict>
+        <OptionInfer>On</OptionInfer>
+    </PropertyGroup>
+
+    <PropertyGroup>
+        <MyType>WindowsForms</MyType>
+    </PropertyGroup>
+ 
+    <ItemGroup>
```
Comment (davkean) [inline src/System.Windows.Forms/tests/IntegrationTests/VisualBasic AppFramework Net5/VbWinForms.vbproj:23]: These imports should be defaulted in the SDK. in the same way others are.

**Comment by davkean** [comment]
@KlausLoeffelmann Bear in mind that we haven't lit up the application pages yet, the VS work to do this isn't yet scheduled @jjmew.

**Comment by davkean** [comment]
@KlausLoeffelmann I had a bunch of feedback on this PR - are there plans to address/file bugs about this?

File: src/System.Windows.Forms/tests/IntegrationTests/VisualBasic AppFramework Net5/My Project/AssemblyInfo.vb (hunk 6 above)
Comment (KlausLoeffelmann) [reply src/System.Windows.Forms/tests/IntegrationTests/VisualBasic AppFramework Net5/My Project/AssemblyInfo.vb:1]: Damn, I missed you comments before a merged.

@KathleenDollard and I were talking about this this morning, and we were not sure if the GUID is needed in context of the SingleInstance functionality.

File: src/System.Windows.Forms/tests/IntegrationTests/VisualBasic AppFramework Net5/VbWinForms.vbproj (hunk 7 above)
Comment (KlausLoeffelmann) [reply src/System.Windows.Forms/tests/IntegrationTests/VisualBasic AppFramework Net5/VbWinForms.vbproj:9]: If we don't have LangVersion here, we're getting a compiler error.

File: src/System.Windows.Forms/tests/IntegrationTests/VisualBasic AppFramework Net5/VbWinForms.vbproj (hunk 9 above)
Comment (KlausLoeffelmann) [reply src/System.Windows.Forms/tests/IntegrationTests/VisualBasic AppFramework Net5/VbWinForms.vbproj:5]: Will correct in a separate PR.

File: src/System.Windows.Forms/tests/IntegrationTests/VisualBasic AppFramework Net5/My Project/AssemblyInfo.vb (hunk 6 above)
Comment (davkean) [reply src/System.Windows.Forms/tests/IntegrationTests/VisualBasic AppFramework Net5/My Project/AssemblyInfo.vb:1]: Regardless of whether it was, we should be adding additional attributes and not turning off the auto-gen of the other attributes.

File: src/System.Windows.Forms/tests/IntegrationTests/VisualBasic AppFramework Net5/VbWinForms.vbproj (hunk 7 above)
Comment (davkean) [reply src/System.Windows.Forms/tests/IntegrationTests/VisualBasic AppFramework Net5/VbWinForms.vbproj:9]: This would be a bug against the Roslyn team, no LangVersion should be present in the project file.

Comment (paul1956) [reply src/System.Windows.Forms/tests/IntegrationTests/VisualBasic AppFramework Net5/VbWinForms.vbproj:9]: I don't know why but VB code in Roslyn does not support version 16 features (without setting language version) but using the same version of Visual Studio, for example Version 16.7.0 Preview 4.0, on any other project VB defaults to version 16. I just say this yesterday while writing a Roslyn test and not was not able to understand where the issue actually is.

File: src/System.Windows.Forms/tests/IntegrationTests/VisualBasic AppFramework Net5/VbWinForms.vbproj (hunk 8 above)
Comment (paul1956) [reply src/System.Windows.Forms/tests/IntegrationTests/VisualBasic AppFramework Net5/VbWinForms.vbproj:12]: In a non-Roslyn project LangVersion defaults to 16 with Visual Studio Version 16.7.0 Preview 4.0. In Roslyn you get the following

image

File: src/System.Windows.Forms/tests/IntegrationTests/VisualBasic AppFramework Net5/My Project/AssemblyInfo.vb (hunk 6 above)
Comment (paul1956) [reply src/System.Windows.Forms/tests/IntegrationTests/VisualBasic AppFramework Net5/My Project/AssemblyInfo.vb:1]: The GUID can't be changed during auto generate on each change of some attribute it is a constant once the project is first created. See other comment about keeping generated ApplicationInfo in source control.

File: src/System.Windows.Forms/tests/IntegrationTests/VisualBasic AppFramework Net5/VbWinForms.vbproj (hunk 8 above)
Comment (davkean) [reply src/System.Windows.Forms/tests/IntegrationTests/VisualBasic AppFramework Net5/VbWinForms.vbproj:12]: This is a bug and should not be worked around in the template.

File: src/System.Windows.Forms/tests/IntegrationTests/VisualBasic AppFramework Net5/VbWinForms.vbproj (hunk 7 above)
Comment (davkean) [reply src/System.Windows.Forms/tests/IntegrationTests/VisualBasic AppFramework Net5/VbWinForms.vbproj:9]: This is a bug and should not be worked around in the template.

File: src/System.Windows.Forms/tests/IntegrationTests/VisualBasic AppFramework Net5/My Project/AssemblyInfo.vb (hunk 6 above)
Comment (davkean) [reply src/System.Windows.Forms/tests/IntegrationTests/VisualBasic AppFramework Net5/My Project/AssemblyInfo.vb:1]: This template is SDK-style and hence should be using auto-generated assembly info. If there is an attribute that can't be represented by a property in the project file, then only this exact attribute should be added this file.

File: src/System.Windows.Forms/tests/IntegrationTests/VisualBasic AppFramework Net5/VbWinForms.vbproj (hunk 8 above)
Comment (paul1956) [reply src/System.Windows.Forms/tests/IntegrationTests/VisualBasic AppFramework Net5/VbWinForms.vbproj:12]: Fixed in #46170 when accepted/Merged

Comment (davkean) [reply src/System.Windows.Forms/tests/IntegrationTests/VisualBasic AppFramework Net5/VbWinForms.vbproj:12]: As mentioned here on https://github.com/dotnet/roslyn/issues/46157, this isn't addressed in the PR.
